lib/samson/retry.rb

```diff
@@ -5,12 +5,16 @@ def self.retry_when_not_unique(&block)
       with_retries [ActiveRecord::RecordNotUnique], 1, &block
     end
 
-    def self.with_retries(errors, count, options = nil)
-      check = options && options[:if]
+    def self.with_retries(errors, count, only_if: nil, sleep: nil)
       yield
     rescue *errors
       count -= 1
-      count >= 0 && (!check || check.call($!)) ? retry : raise
+      if count >= 0 && (!only_if || only_if.call($!))
+        sleep(sleep) if sleep
```

Comment:
super confusing variable names here, maybe name the argument `sleep_duration` or `wait_time` or something?

Reply:
looks so neat though ... changed 😿